src/edu/washington/escience/myria/operator/Limit.java

```diff
@@ -38,23 +41,30 @@ public Limit(@Nonnull final Long limit, final Operator child) {
     this.limit = Objects.requireNonNull(limit, "limit");
     Preconditions.checkArgument(limit >= 0L, "limit must be non-negative");
     toEmit = this.limit;
+    done = false;
```

Comment:
I don't think you need this new flag. You should immediately return `null` from `fetchNextReady()` if the child has already been closed, which you can test with `child.isOpen()` or `child.eos()`. @jingjingwang do you agree?